pkgs/build-support/docker/default.nix

```diff
@@ -250,6 +253,12 @@ rec {
 
       ${postUmount}
       '');
+    in
+    runCommand name {} ''
+      mkdir -p $out
+      cd ${result}
+      cp layer.tar json VERSION $out
+    '';
```

Comment:
Could you clarify the purpose of this `runCommand` wrapper?
It was found to be broken: https://github.com/NixOS/nixpkgs/pull/80068/files#r613275987

Reply:
If the goal is to reduce output size, it's generally better to do it in the original derivation. Doing so reduces the need for garbage collection, reduces I/O and reduces the storage required in cases where everything is cached eagerly.

Reply:
How does this reduce output size, if I just want to extract some files from a container image? I'll need to create a tar, just to unpack it again in the next build step.

Reply:
It seems to have very little utililty. The only role I can think of for this `runCommand` call is to keep redundant files from propagating to other derivations, but that should probably be done in `result` itself instead.

@srhb is there anything to this code that we're not thinking of?

Reply:
I'm fairly confident in my guess that it was to remove the files created by `createEmptyDiskImage`. I've made it write to a different location so we don't need this extra derivation anymore. Done in https://github.com/NixOS/nixpkgs/pull/80068

Reply:
I realize now that I asked the wrong person about these lines. Should've been @dingxiangfei2009 
Oh well.